awkernel_async_lib/src/scheduler/gedf.rs

```diff
@@ -184,3 +191,37 @@ impl GEDFScheduler {
         false
     }
 }
+
+pub fn calculate_and_update_dag_deadline(dag_info: &DagInfo, wake_time: u64) -> u64 {
+    let dag_id = dag_info.dag_id;
+    let node_index = dag_info.node_index;
+
+    if let Some(absolute_deadline) = get_dag_absolute_deadline(dag_id) {
+        let dag = get_dag(dag_id).expect("GEDF scheduler: DAG {dag_id} not found");
+        let current_node_index = to_node_index(node_index);
```

Comment:
How about directly using `NodeIndex` type instead of `u32` for `DagInfo.node_index` without `to_node_index()`?